### PR TITLE
Calculate hashCode with arguments that are consistent with equals

### PR DESCRIPTION
**Pull request by AniketNS**
## Calculate hashCode with arguments that are consistent with equals

I used `String.format()` function for better consistency and it is more understandable.

Instead of relying on the `hashCode` of the superclass I've used the `Object.hash()` method to generate a hashcode. I think it is easier to understand and more explicit.

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/jenkinsci/git-client-plugin/blob/master/CONTRIBUTING.adoc) doc
- [ ] I have referenced the Jira issue related to my changes in one or more commit messages
- [ ] I have added tests that verify my changes
- [x] Unit tests pass locally with my changes
- [x] I have added documentation as necessary
- [x] No Javadoc warnings were introduced with my changes
- [x] No spotbugs warnings were introduced with my changes
- [ ] I have interactively tested my changes

## Types of changes

What types of changes does your code introduce?

- [x] Infrastructure change (non-breaking change which updates dependencies or improves infrastructure)

### CODE REVIEW

File: src/main/java/hudson/plugins/git/Branch.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public String toString() {
      */
     @Override
     public int hashCode() {
-        return super.hashCode();
+        return Objects.hash(name, sha1);
```
Comment (MarkEWaite) [inline src/main/java/hudson/plugins/git/Branch.java:53]: I disagree with this choice.  The `Branch` object inherits a name field and a sha1 field from the parent `GitObject`.  By replacing the `super.hashCode()` with ` Objects.hash(name, sha1)`, it is unnecessarily inserting detailed knowledge of the relevant fields of the `GitObject` into the `Branch` object.  If the `GitObject` were to add relevant fields in the future, then this `hashCode` method would need to be changed as well.

Can you explain further why you think this is a positive change?
```suggestion
        return super.hashCode();
```

**Comment by MarkEWaite** [comment]
@AniketNS thanks for the pull request.  Please do the following:

* [ ] Restore the pull request template.  The template is placed in this repository so that every pull request submitter sees the expectations of the maintainers and clearly states which items they were able to complete and which they were not able to complete.  Pull request templates are widely used in the Jenkins project so that contributors see the expectations of maintainers while creating the pull request
* [ ] Describe the changes more clearly in the pull request summary.  A Jenkins user that reads the changelog for the git client plugin will not understand the change based on the description, "Made changes in Branch.java".  Why were those changes made and how do they improve things for the user or for the plugin maintainers or for someone else?

**Review by MarkEWaite — CHANGES_REQUESTED** [review]

File: src/main/java/hudson/plugins/git/Branch.java (hunk 1 above)
Comment (AniketNS) [reply src/main/java/hudson/plugins/git/Branch.java:53]: Your disagreement raises a valid point, and I appreciate your thoughtful consideration. But @MarkEWaite don't you think that `hashCode` method depends upon the `GitObject.java` file itself a wrong practice? 

That's why I wanted to make this `hashCode` method independent. By explicitly specifying the fields used for hashing (name and sha1), it avoids potential inconsistencies that might arise if the superclass's `hashCode` logic were to change.

Comment (MarkEWaite) [reply src/main/java/hudson/plugins/git/Branch.java:53]: I think that you're right.  The `equals` method in the class already includes the knowledge that `name` and `sha1` are the significant fields of the class.  Using `Objects.hash()` makes the `hashCode` method consistent with the `equals` method in the class.  That seems like a valuable and helpful consistency.

**Comment by AniketNS** [comment]
Hello @MarkEWaite, could you please provide some info on the current status of this PR? I'm wondering if there are any specific improvements or additional changes I should make. I appreciate your feedback.

**Comment by MarkEWaite** [comment]
> Hello @MarkEWaite, could you please provide some info on the current status of this PR? I'm wondering if there are any specific improvements or additional changes I should make. I appreciate your feedback.

Please take the actions that I listed in the checklist of the [earlier comment](https://github.com/jenkinsci/git-client-plugin/pull/1097#issuecomment-1892402916).  those actions include:

* [ ] Restore the pull request template.  The template is placed in this repository so that every pull request submitter sees the expectations of the maintainers and clearly states which items they were able to complete and which they were not able to complete.  Pull request templates are widely used in the Jenkins project so that contributors see the expectations of maintainers while creating the pull request
* [ ] Describe the changes more clearly in the pull request summary.  A Jenkins user that reads the changelog for the git client plugin will not understand the change based on the description, "Made changes in Branch.java".  Why were those changes made and how do they improve things for the user or for the plugin maintainers or for someone else?

**Comment by AniketNS** [comment]
Hey @MarkEWaite, I've changed the Pull request name but can you please tell me now how I can restore the pull request template? 

Even if I copy the content from [this](https://github.com/jenkinsci/git-client-plugin/pull/1099) pull req template then how can I add that checklist?

**Comment by MarkEWaite** [comment]
> Even if I copy the content from [this](https://github.com/jenkinsci/git-client-plugin/pull/1099) pull req template then how can I add that checklist? Or should I close this pull request and open a new one with the default pull request template and with a clear pull request description as you told me before?

One way is to copy the [source text of the pull request template](https://github.com/jenkinsci/git-client-plugin/blob/master/.github/pull_request_template.md) and paste it into the description of this pull request.  The source text is also on your local copy of the repository if you prefer to copy it from a local text file.

**Comment by AniketNS** [comment]
Thank you so much @MarkEWaite. I've made the changes as requested. Please check and let me know if I've missed anything.

**Review by MarkEWaite — APPROVED** [review]

**Pull request merged**